File: app/src/main/java/com/lz233/onetext/tools/utils/AppUtil.java
package com.lz233.onetext.tools.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class AppUtil {
    public static boolean isUseWifi(Context context) {
        ConnectivityManager connMgr = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connMgr.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
        Boolean isWifiConn = networkInfo.isConnected();
        return isWifiConn;
    }

    public static int dp2px(Context context, float dipValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dipValue * scale + 0.5f);
    }

    public static int px2sp(Context context, float pxValue) {
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (pxValue / fontScale + 0.5f);
    }

    public static String dateToStamp(String time) throws ParseException {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        Date date = simpleDateFormat.parse(time);
        long ts = date.getTime();
        return String.valueOf(ts);
    }
}

File: app/src/main/java/com/lz233/onetext/tools/utils/AppUtil.kt
package com.lz233.onetext.tools.utils

import android.content.Context
import android.net.ConnectivityManager
import java.text.SimpleDateFormat

fun isUseWifi(context: Context): Boolean {
    val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
    val networkInfo = connMgr.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
    return networkInfo?.isConnected ?: true
}

fun dp2px(context: Context, dipValue: Float): Int = (dipValue * context.resources.displayMetrics.density + 0.5f).toInt()

fun px2sp(context: Context, pxValue: Float): Int = (pxValue / context.resources.displayMetrics.scaledDensity + 0.5f).toInt()

fun dateToStamp(time: String): String {
    val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
    val date = simpleDateFormat.parse(time)
    val ts = date?.time
    return ts.toString()
}